bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/DoorbirdHandlerFactory.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.doorbird.internal;
+
+import static org.openhab.binding.doorbird.internal.DoorbirdBindingConstants.*;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.smarthome.core.i18n.TimeZoneProvider;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory;
+import org.eclipse.smarthome.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * The {@link DoorbirdHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+@Component(configurationPid = "binding.doorbird", service = ThingHandlerFactory.class)
+public class DoorbirdHandlerFactory extends BaseThingHandlerFactory {
+    private @NonNullByDefault({}) TimeZoneProvider timeZoneProvider;
```

Comment:
Have a look at constructor injection (e.g. SNMP binding). You can remove the getters, setters and the annotations here.

Reply:
I converted to use constructor injection, but I'm having trouble building the binding now. It seems to be having an issue with `HttpClient`.

```
[ERROR] Failed to execute goal org.apache.karaf.tooling:karaf-maven-plugin:4.2.6:verify (karaf-feature-verification) on project org.openhab.binding.doorbird: Feature resolution failed for [openhab-binding-doorbird/2.5.0.SNAPSHOT]
[ERROR] Message: Unable to resolve root: missing requirement [root] osgi.identity; osgi.identity=openhab-binding-doorbird; type=karaf.feature; version=2.5.0.SNAPSHOT; filter:="(&(osgi.identity=openhab-binding-doorbird)(type=karaf.feature)(version>=2.5.0.SNAPSHOT))" [caused by: Unable to resolve openhab-binding-doorbird/2.5.0.SNAPSHOT: missing requirement [openhab-binding-doorbird/2.5.0.SNAPSHOT] osgi.identity; osgi.identity=org.openhab.binding.doorbird; type=osgi.bundle; version="[2.5.0.201908231958,2.5.0.201908231958]"; resolution:=mandatory [caused by: Unable to resolve org.openhab.binding.doorbird/2.5.0.201908231958: missing requirement [org.openhab.binding.doorbird/2.5.0.201908231958] osgi.service; filter:="(objectClass=org.eclipse.jetty.client.HttpClient)"; effective:=active]]
[
```

This is the relevant part of the handler factory code. Am I missing something obvious?

```
@NonNullByDefault
@Component(configurationPid = "binding.doorbird", service = ThingHandlerFactory.class)
public class DoorbirdHandlerFactory extends BaseThingHandlerFactory {
    private final TimeZoneProvider timeZoneProvider;
    private final HttpClient httpClient;

    @Activate
    public DoorbirdHandlerFactory(@Reference TimeZoneProvider timeZoneProvider, @Reference HttpClient httpClient) {
        this.timeZoneProvider = timeZoneProvider;
        this.httpClient = httpClient;
    }
```

Reply:
Nevermind. I got it sorted out. This is what I needed.

```
    @Activate
    public DoorbirdHandlerFactory(@Reference TimeZoneProvider timeZoneProvider,
            @Reference HttpClientFactory httpClientFactory) {
        this.timeZoneProvider = timeZoneProvider;
        this.httpClient = httpClientFactory.getCommonHttpClient();
    }
```